tests/test_replay_buffer.py

```diff
@@ -487,6 +487,101 @@ def test_append_and_sample(self):
                     self.assertEqual(t0['next_action'], t1['action'])
 
 
+@testing.parameterize(*testing.product({
+    'replay_buffer_type': ['ReplayBuffer', 'PrioritizedReplayBuffer'],
+}))
+class TestReplayBufferWithEnvID(unittest.TestCase):
+
+    def test(self):
+        n = 5
+        if self.replay_buffer_type == 'ReplayBuffer':
+            rbuf = replay_buffer.ReplayBuffer(capacity=None, num_steps=n)
+        elif self.replay_buffer_type == 'PrioritizedReplayBuffer':
+            rbuf = replay_buffer.PrioritizedReplayBuffer(
+                capacity=None, num_steps=n)
+        else:
+            assert False
+
+        # 2 transitions for env_id=0
+        for i in range(2):
+            trans1 = dict(state=0, action=1, reward=2, next_state=3,
+                          next_action=4, is_state_terminal=False)
+            rbuf.append(env_id=0, **trans1)
+        # 4 transitions for env_id=1 with a terminal state
+        for i in range(4):
+            trans1 = dict(state=0, action=1, reward=2, next_state=3,
+                          next_action=4, is_state_terminal=(i == 3))
+            rbuf.append(env_id=1, **trans1)
+        # 9 transitions for env_id=2
+        for i in range(9):
+            trans1 = dict(state=0, action=1, reward=2, next_state=3,
+                          next_action=4, is_state_terminal=False)
+            rbuf.append(env_id=2, **trans1)
+
+        # It should have:
+        #   - 4 transitions from env_id=1
+        #   - 5 transitions from env_id=2
+        self.assertEqual(len(rbuf), 9)
```

Comment:
Shouldn't we have 3 transitions? One from `env_id=1` and five from `env_id=2`?

Reply:
Four form `env_id=1`: (s_0, s_4), (s_1, s_4), (s_2, s_4), and (s_3, s_4).

Reply:
Got it.